usage.jenkins.io.tf

```diff
@@ -0,0 +1,32 @@
+resource "digitalocean_ssh_key" "usage_jenkins_io" {
+  name = "Administrator Public SSH Key for usage.jenkins.io"
+  # the private key is stored within sops: config/usage.jenkins.io/id_do_usage_jenkins_io
+  public_key = file("ssh/id_do_usage_jenkins_io.pub")
+}
+
+resource "digitalocean_volume" "usage_jenkins_io_data" {
+  region                  = var.region
+  name                    = "usagejenkinsiodata" # Only lowercase alphanum
+  size                    = 750                  # will be extended in a second time to check online extend
+  initial_filesystem_type = "ext4"
+  description             = "Data disk for usage.jenkins.io"
+}
+
+resource "digitalocean_volume_attachment" "usage_jenkins_io_data" {
+  droplet_id = digitalocean_droplet.usage_jenkins_io.id
+  volume_id  = digitalocean_volume.usage_jenkins_io_data.id
+}
+
+resource "digitalocean_droplet" "usage_jenkins_io" {
+  # default username is root. TODO change it with cloudinit
+  image       = "ubuntu-22-04-x64"
+  name        = "usage.jenkins.io"
+  region      = var.region
+  size        = "s-2vcpu-4gb"
+  monitoring  = true
+  ipv6        = true
+  resize_disk = true
+  ssh_keys    = [digitalocean_ssh_key.usage_jenkins_io.fingerprint]
+  user_data   = templatefile("${path.root}/cloudinit.tftpl", { hostname = "do.usage.jenkins.io" })
```

Comment:
As seen together during pairing review: we want to have a dedicated TPL file for usage here.
Because as per https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/droplet#user_data-1 , changing the user data will destroy and re-create VMs. Which we do not want: lets ensure it is flexible for us in the future.

=> we want to use the content copied from the "current" tpl in https://github.com/jenkins-infra/shared-tools/blob/main/terraform/cloudinit.tftpl. We only have to remove the 2 user/group ID/GID change (only needed in Azure / AWS).
=> of course, we'll have one file per VM here (we'll have to move archives's one in a subseqsuent PR eventually)